src/java/ognl/OgnlRuntime.java

```diff
@@ -899,12 +913,53 @@ public static Object invokeMethod(Object target, Method method, Object[] argsArr
                 }
             }
 
-            result = method.invoke(target, argsArray);
+            result = invokeMethodInJavaSandbox(target, method, argsArray);
         }
 
         return result;
     }
 
+    private static Object invokeMethodInJavaSandbox(Object target, Method method, Object[] argsArray) throws InvocationTargetException, IllegalAccessException {
+        SecurityManager parentSecurityManager = System.getSecurityManager();
```

Comment:
Why not to use `this. _securityManager`? Someone could already re-define the SecurityManager used by OGNL (OGNL already supports checking permissions)

Reply:
Yes I saw, but with `this. _securityManager` it seems user only can say OGNL if invocation of e.g. `some.package.SomeClass.someMethod` is permitted or not. OK it's nice and I kept it.

But my security manager *then in a higher level* guarantees that any invocation *doesn't reach* top most dangerous actions (e.g. execute,exit,thread access and etc) even if e.g. an inner of inner of inner method is going to do that action. It's an awesome JDK security model that guarantees this! For example if an attacker was successful to inject `someField=new some.third.party.package.that.target.app.uses.SomeClass().getScriptEngine().run('cat pwd')` then my security manager doesn't allow this (even if user has forgot to add this to blacklist!) because JDK security model detects that it finally reaches an execution and throws an exception for me 😃 

At bottom, despite it seems a massive change, I think it's fine and OK to add it to all branches because firstly it's really a simple but great security improvement IMHO and secondly if it breaks something in user apps, then they will be happy to discover because that thing is a security flaw - my security manager only blocks top most dangerous actions (see overloaded methods) and relaxes with others and user's apps must already satisfy this rule.